### PR TITLE
Fix: spring cloud error and refactor event callback

### PR DESCRIPTION
**Pull request by PhilYue**
**What this PR does**:
1. fix spring cloud error that "zookeeper node not exist or some other"
2. add zookeeper root path config for spring cloud change the discovery root, like "/services"
3. refactor the watch event handle

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go [hunk 1]
```diff
@@ -124,29 +119,43 @@ func (z *zkAppListener) watchEventHandle(children []string, e <-chan zk.Event) b
...
-	del := func() {
-		keys := Keys(discovery.getServiceMap())
-		diff := Diff(keys, fetchChildren)
-		if diff != nil {
-			logger.Debugf("Del the service %s", diff)
-			for _, sn := range diff {
-				for _, instance := range discovery.getServiceMap()[sn] {
-					discovery.delServiceInstance(instance)
-				}
+		if !contains(fetchChildren, sn) {
```
Comment (mark4z) [inline pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go:140]: fix this.

File: pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go [hunk 2]
```diff
@@ -124,29 +119,43 @@ func (z *zkAppListener) watchEventHandle(children []string, e <-chan zk.Event) b
...
 			}
 		}
 	}
-	del()
 
 	for _, serviceName := range fetchChildren {
 		serviceNodePath := strings.Join([]string{z.servicesPath, serviceName}, constant.PathSlash)
+		instances := serviceMap[serviceName]
+		if instances == nil || len(instances) == 0 {
+			z.svcListeners.RemoveListener(serviceNodePath)
```
Comment (mark4z) [inline pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go:157]: Same

**Review by mark4z — APPROVED** [review]

File: pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go (hunk 1 above)
Comment (PhilYue) [reply pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go:140]: done

File: pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go (hunk 2 above)
Comment (PhilYue) [reply pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go:157]: done

**Review by AlexStocks — APPROVED** [review]

**Review by ztelur — APPROVED** [review]

**Pull request merged**